docs/chapters/crud/index.md

```diff
@@ -0,0 +1,49 @@
+# CRUD Operations
+
+CRUD stands for Create, Read, Update, Delete. Together, they form the basis of any database engine. While Jina itself is *not* a database engine, it does provide these methods of interacting with the data stored in the indexers.
+
+Until now, Jina has only supported indexing (creating) and querying (reading) documents. To update or delete a document, you'd have to edit your dataset, and then re-build the Flow and indexers. Needless to say, this could create problems if you had large datasets.
+
+With the release of version `1.0` we're introducing **update** and **delete** operations. These are implemented across our executors and drivers and will allow you to update and delete documents by their ids. 
+
+A basic example of this can be found in the `test_crud.py` file under `tests/integration/crud/simple`:
+
+The Flow object now supports a `delete` and `update` method, with a signature similar to index:
+
+```python
+    with f:
+        f.index(input_fn=random_docs(10))
+
+    with f:
+        f.update(input_fn=random_docs(10))
+
+    with f:
+        f.delete(input_fn=random_docs(10))
+```
+
+Note: deletion and update will happen by `id` of the document.
+
+## Limitations
+
+Unfortunately there are some limitations to what Jina can do for the moment. These were trade-offs we needed to implement to keep Jina performant, robust, and easy to use. Some of these will be addressed in future versions, while some are intrinsic to Jina's architecture.
+
+1. **Partial update**
+
+For the moment we do not support partial updates. So if you want to update a document, you need to send the entire document. This is due to Jina's architecture: the document is sent as one piece through the Flow.
+
+1. **Update flows**
+
+In the context of Flows with segmenters and chunks, a Document may end up being split into chunks. Currently, the Update request will not work for these. You will need to manually remove the chunks by their `ids`. Then you can remove the parent document as well, by its `id`. Finally, you can index the new document, with its new contents (and thus new, different child chunks).
+
+1. **Sharding**
+
+When sharding data in indexers, the data will be split across these. This is achieved due to the `polling: any` configuration. During a query, you will need to set `polling: all`. However, this will lead to some shards getting a query request with a key that doesn't exist. In this case, warnings will be emitted by the indexer. You can ignore these within this context. The warnings are there for the situations when missing keys are *not* expected.
+
+1. **Advanced Indexers**
+
+`Faiss` & `Annoy` indexers don't support updates while querying. You can still use them in query mode, but you will not be able to add, delete, or update data. In order to do this, you will need use a `ref_indexer`-based approach. An example of this can be found [here](https://github.com/jina-ai/examples/master/advanced-vector-search)
```

Comment:
This is not totally true, I think it works since `Faiss` and `Annoy` only start working while Querying.

Reply:
Since update and delete are operations on BaseNumpyIndexer these operations are possible on Faiss and Annoy instances.

Reply:
Re-wrote this part